### PR TITLE
Add feature to update user profile

### PR DESCRIPTION
**Pull request by tanay13**
files #15 

@YashKumarVerma

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -75,11 +83,24 @@ export class UserController {
...
+    return res.render('profile/edit.hbs',{user})
   }
 
+  @Put('/:id')
+  @UseGuards(JwtAuthGuard)
+  async update(@Param('id') id: string, @Body() updateUserDto: UpdateUserDto,@Res() res,@Request() req) {
+    const user = await this.userService.update(id, updateUserDto);
+    return this.DashboardController.showProfile(req,res)
+  }   
+
```
Comment (YashKumarVerma) [inline src/user/user.controller.ts:103]: Send response only, no need to render.

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]

**Comment by YashKumarVerma** [comment]
@tanay13 ,

the reason that this does not render the profile page is that this is called via a network call.

i.e. the fetch command calls this, and this should only send data as a response. 

If you remove the `.json()` function from the fetch-promise-chain, then you will see that the server actually renders the page.

image

## What you need to do
- either implement this feature as a simple form that responds to `POST` request on the same page.
- or, to avoid much changes in the current codebase, instead of redirecting the user, send a response (only json), and then show a message on the edit page.

File: src/user/user.controller.ts (hunk 1 above)
Comment (tanay13) [reply src/user/user.controller.ts:103]: Can you explain this a bit , I didn't get your point ?

**Comment by tanay13** [comment]
@YashKumarVerma Can you check once again? Instead of redirecting the user to profile page i redirected it to the dashboard . Now its working fine . Please review this .

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]
image

but the edit page shows only the first component of the name

image

**Comment by tanay13** [comment]
@YashKumarVerma please check i fixed it

**Review by YashKumarVerma — COMMENTED** [review]
Awesome Work !

**Comment by YashKumarVerma** [comment]
Thanks a lot for your contributions!

**Pull request merged**